diff/diff.go

```diff
@@ -0,0 +1,293 @@
+package diff
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+// Difference represents a "chunk" of difference between two
+// `Object`s.
+type Difference interface {
+	Summarise(out io.Writer)
+	AtPath() string
+}
+
+type Chunk struct {
+	Deleted []interface{}
+	Added   []interface{}
```

Comment:
To have better type safety, could the added/deleted be strings? Or io.Readers?
Do they need to be interfaces?